dev/integration_tests/ios_add2app/ios_add2appTests/IntegrationTests.m

```diff
@@ -6,12 +6,30 @@
 #import <XCTest/XCTest.h>
 
 #import "../ios_add2app/AppDelegate.h"
-#import "../ios_add2app/MainViewController.h"
+#import "../ios_add2app/DualFlutterViewController.h"
 #import "../ios_add2app/FullScreenViewController.h"
-
-static void waitForInitialFlutterRender() {
-  // TODO(dnfield,jamesderlin): actually sync with Flutter rendering.
-  CFRunLoopRunInMode(kCFRunLoopDefaultMode, 10, false);
+#import "../ios_add2app/MainViewController.h"
+#import "../ios_add2app/HybridViewController.h"
+
+static void waitForFlutterSemanticsTree(FlutterViewController *viewController) {
+  int tries = 10;
+  double delay = 1.0;
+  __block BOOL semanticsAvailable = NO;
+  __block id<NSObject> observer = [[NSNotificationCenter defaultCenter]
+      addObserverForName:@"FlutterSemanticsUpdate"
+                  object:viewController
+                   queue:nil
+              usingBlock:^(NSNotification *notification) {
+                semanticsAvailable = YES;
+                [[NSNotificationCenter defaultCenter] removeObserver:observer];
+              }];
+  [viewController.engine ensureSemanticsEnabled];
+  while (semanticsAvailable == NO && tries != 0) {
```

Comment:
Add one line saying what is synchronous and what isn't in a comment here and in the ensureSemanticsEnabled objective-c doc